src/Microsoft.Azure.WebJobs.Host/Bindings/Data/StructDataBinding.cs

```diff
@@ -9,17 +9,21 @@
 
 namespace Microsoft.Azure.WebJobs.Host.Bindings.Data
 {
+    /// <summary>
+    /// Handles value types (structs) as well as nullable types.
+    /// </summary>
     internal class StructDataBinding<TBindingData> : IBinding
-        where TBindingData : struct
     {
         private static readonly IObjectToTypeConverter<TBindingData> Converter =
             ObjectToTypeConverterFactory.CreateForStruct<TBindingData>();
 
+        private readonly bool _isNullable;
         private readonly string _parameterName;
         private readonly IArgumentBinding<TBindingData> _argumentBinding;
 
         public StructDataBinding(string parameterName, IArgumentBinding<TBindingData> argumentBinding)
         {
+            _isNullable = typeof(TBindingData).IsGenericType && typeof(TBindingData).GetGenericTypeDefinition() == typeof(Nullable<>);
```

Comment:
>typeof(TBindingData).IsGenericType && typeof(TBindingData).GetGenericTypeDefinition() == typeof(Nullable<>); [](start = 26, length = 108)

nit: move to the existing static TypeUtility class: 
  bool TypeUtility.IsNullable(Type)